localstack-core/localstack/aws/handlers/validation.py

```diff
@@ -22,19 +23,23 @@
 
 LOG = logging.getLogger(__name__)
 
-
-# TODO: replace with from importlib.resources.files when https://github.com/python/importlib_resources/issues/311 is
-#   resolved. Import from a namespace package is broken when installing in editable mode.
+# todo: replace with importlib.resources, which currently seems to have problems with plux
 oas_path = os.path.join(os.path.dirname(__file__), "..", "..", "openapi.yaml")
 
 
+def get_oas_spec() -> dict:
```

Comment:
I think we really should try to avoid monkey patches as much as possible. Is there another way to implement this extension mechanism you are planning? Can we get a cleaner interface on the OpenAPI "spec loader"?

Reply:
Thanks for the comment. As we briefly discussed, I reworked this with proper plugins. Core now has its plugin and pro will define its own. The validators can now work with multiple `openapi` instances to perform the validation.